src/arch/arm/insts/static_inst.cc

```diff
@@ -599,6 +599,16 @@ ArmStaticInst::printDataInst(std::ostream &os, bool withImm,
         firstOp = false;
         printIntReg(os, rd);
     }
+    if (rd == INTREG_ZERO) { // Print zero register as destination for SUBS (immediate).
+        std::stringstream ss;
+        ss << os.rdbuf();
+        std::string str(ss.str());
+        int pos = str.find("subs");
+        if (pos != std::string::npos) {
+            firstOp = false;
+            printIntReg(os, rd);
```

Comment:
Maybe you should change it to INTREG_X31 here instead of modifying the above code block.

Reply:
And this change looks not the perfect way of fixing the problem, so IMO, it won't get merged by upstream.
However it may not introduce new problems to our project.